src/app/course/course-list/course-list.component.ts

```diff
@@ -1,15 +1,29 @@
-import { Component, OnInit } from '@angular/core';
-
-@Component({
-  selector: 'app-course-list',
-  templateUrl: './course-list.component.html',
-  styleUrls: ['./course-list.component.scss']
-})
-export class CourseListComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit(): void {
-  }
-
-}
+import { Component, OnInit } from '@angular/core';
+import { COURSES } from '../mock-data';
+
+@Component({
+  selector: 'app-course-list',
+  templateUrl: './course-list.component.html',
+  styleUrls: ['./course-list.component.scss']
+})
+export class CourseListComponent implements OnInit {
+  courseList = COURSES;
+
+  ngOnInit(): void {
+    console.log('OnInit')
+  }
+
+
+  public deleteCourse(item): void {
+    console.log('this course was delete:', item.id)
+  }
+
+  public editCourse(item): void {
+    console.log('this course was edite:', item.id)
+  }
+
+  public trackByFn(index: number): number {
+    return index;

```

Comment:
That's fine for a homework, but keep in mind that the index-based approach may not be a good choice for some cases. In production, each `*ngFor` should be analyzed to choose a right solution